src/dowel/csv_output.py

```diff
@@ -19,42 +21,51 @@ def __init__(self, file_name):
         self._fieldnames = None
         self._warned_once = set()
         self._disable_warnings = False
+        self.tabular = TabularInput()
 
     @property
     def types_accepted(self):
-        """Accept TabularInput objects only."""
-        return (TabularInput, )
-
-    def record(self, data, prefix=''):
-        """Log tabular data to CSV."""
-        if isinstance(data, TabularInput):
-            to_csv = data.as_primitive_dict
-
-            if not to_csv.keys() and not self._writer:
-                return
-
-            if not self._writer:
-                self._fieldnames = set(to_csv.keys())
-                self._writer = csv.DictWriter(
-                    self._log_file,
-                    fieldnames=self._fieldnames,
-                    extrasaction='ignore')
-                self._writer.writeheader()
-
-            if to_csv.keys() != self._fieldnames:
-                self._warn('Inconsistent TabularInput keys detected. '
-                           'CsvOutput keys: {}. '
-                           'TabularInput keys: {}. '
-                           'Did you change key sets after your first '
-                           'logger.log(TabularInput)?'.format(
-                               set(self._fieldnames), set(to_csv.keys())))
-
-            self._writer.writerow(to_csv)
-
-            for k in to_csv.keys():
-                data.mark(k)
-        else:
-            raise ValueError('Unacceptable type.')
+        """Accept str and scalar objects."""
+        return (str, ) + np.ScalarType
+
+    @property
+    def keys_accepted(self):
```

Comment:
i think this should be a constructor arg for all log outputs. people might want to make loggers which subscribe to different streams of keys (this is one of the nice parts of using keys).